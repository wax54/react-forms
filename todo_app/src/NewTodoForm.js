import React, { useState } from "react";

const NewTodoForm = ({ onSubmit }) => {
    const INITIAL_STATE = "";
    const [title, setTitle] = useState(INITIAL_STATE);
    const handleSubmit = evt => {
        evt.preventDefault();
        onSubmit(title);
        setTitle(INITIAL_STATE);
    }

    const handleChange = (evt) => setTitle(evt.target.value);
    
    return (
        <form className="NewTodoForm" onSubmit={handleSubmit}>
            <label htmlFor="title-input">Task: </label>
            <input 
                id="title-input"
                type="text" 
                name="title"
                value={title}
                onChange={handleChange}
                placeholder="Wash Dog"
                required
            />
            <button>Add Todo</button>

        </form>
    )
};
export default NewTodoForm