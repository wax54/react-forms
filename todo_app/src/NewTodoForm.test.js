import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from './NewTodoForm';

it("Doesn't Blow up", () => {
    render(<NewTodoForm />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});


it("should allow users to type into the inputs", () => {
    const testTitle = "NEW TASK";

    const { getByLabelText } = render(<NewTodoForm onSubmit={() => undefined} />);

    const titleInput = getByLabelText("Task", { exact: false });
    expect(titleInput.value).toEqual("");
    fireEvent.change(titleInput, { target: { value: testTitle } });
    expect(titleInput.value).toEqual(testTitle);
});

it("should pass input data to the passed function when submited", () =>{
    const testTitle = "NEW TASK";

    let submittedTitle;

    const mockedOnSubmit = title => submittedTitle = title;


    const { getByLabelText, getByText } = render(<NewTodoForm onSubmit={mockedOnSubmit} />);

    const titleInput = getByLabelText("Task", { exact: false });
    fireEvent.change(titleInput, { target: { value: testTitle } });

    const btn = getByText("Add Todo", { exact: false });

    fireEvent.click(btn);

    expect(submittedTitle).toEqual(testTitle);
});


