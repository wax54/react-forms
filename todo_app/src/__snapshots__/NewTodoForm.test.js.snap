// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="title-input"
    >
      Task: 
    </label>
    <input
      id="title-input"
      name="title"
      placeholder="Wash Dog"
      required=""
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
